lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -1107,6 +1107,10 @@ def install_puppet_agent_dev_repo_on( hosts, opts )
             onhost_copy_base = opts[:copy_dir_external]
 
             case variant
+            when /^eos/
```

Comment:
Not for this PR, but @kevpl it wouldn't be the worst thing in the world for a set of platform / variant predicate methods to exist which wrap up our regexes and centralize them.  Something like `variant_is_eos?` (e.g.).  I may need a refresher course on when we should look to `variant` vs. `host['platform']` and what the differences are.